app/src/main/kotlin/org/fossify/smsmessenger/activities/ConversationDetailsActivity.kt

```diff
@@ -60,6 +74,62 @@ class ConversationDetailsActivity : SimpleActivity() {
         binding.membersHeading.setTextColor(primaryColor)
     }
 
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun setupCustomNotifications() {
+        binding.apply {
+            notificationsHeading.beVisible()
+            customNotificationsHolder.beVisible()
+            customNotifications.isChecked = config.customNotifications.contains(threadId.toString())
+            customNotificationsButton.beVisibleIf(customNotifications.isChecked)
+
+            customNotificationsHolder.setOnClickListener {
+                customNotifications.toggle()
+                if (customNotifications.isChecked) {
+                    customNotificationsButton.beVisible()
+                    config.addCustomNotificationsByThreadId(threadId)
+                    createNotificationChannel()
+                } else {
+                    customNotificationsButton.beGone()
+                    config.removeCustomNotificationsByThreadId(threadId)
+                    removeNotificationChannel()
+                }
+            }
+
+            customNotificationsButton.setOnClickListener {
+                Intent(Settings.ACTION_CHANNEL_NOTIFICATION_SETTINGS).apply {
+                    putExtra(Settings.EXTRA_APP_PACKAGE, packageName)
+                    putExtra(Settings.EXTRA_CHANNEL_ID, threadId.hashCode().toString())
+                    startActivity(this)
+                }
+            }
+        }
+    }
+
+    private fun getNotificationManager() = getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun createNotificationChannel() {
+        val name = conversation?.title
+        val audioAttributes = AudioAttributes.Builder()
+            .setUsage(AudioAttributes.USAGE_NOTIFICATION)
+            .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+            .setLegacyStreamType(AudioManager.STREAM_NOTIFICATION)
+            .build()
+
+        NotificationChannel(threadId.hashCode().toString(), name, NotificationManager.IMPORTANCE_HIGH).apply {
```

Comment:
@Aga-C Is there any reason we are using `threadId.hashCode().toString()` and not `threadId.toString()`?

Reply:
Changed it to just `threadId.toString()` as mapping to Int is not required in this case. The notification IDs is required to be an Int.